genienlp/tasks/almond/__init__.py

```diff
@@ -288,9 +292,27 @@ def _make_example(self, parts, dir_name=None, **kwargs):
         return Example.from_raw(self.name + '/' + _id, context, question, answer,
                                 preprocess=self.preprocess_field, lower=False)
 
+# TODO add a similar preprocessing step to Multilingual dialogue tasks as well
+class BaseAlmondDialogueNLUTask(BaseAlmondTask):
+    def preprocess_field(self, sentence, field_name=None):
+        # remove the $ at the start of dialogue
+        # this is safe because we know we're processing dialogues, so the answer
+        # always starts with $dialogue and the context is either `null` or also
+        # starts with $dialogue
+        if field_name == 'context' and sentence.startswith('$dialogue @org.thingpedia.dialogue.transaction . '):
+            sentence = sentence[len('$dialogue @org.thingpedia.dialogue.transaction . '):]
+        if field_name == 'answer':
+            assert(sentence.startswith('$dialogue @org.thingpedia.dialogue.transaction . '))
+            sentence = sentence[len('$dialogue @org.thingpedia.dialogue.transaction . '):]
```

Comment:
Removing `$dialogue` is safe, removing `@org.thingpedia.dialogue.transaction` is not safe, because there might other dialogue policies in the future. Also, if you remove `$dialogue` entirely, you should update the comment.

Reply:
OK, I changed it to `$dialogue`.